internal/config/aws/credentials.go

```diff
@@ -0,0 +1,49 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+type AwsCredentials struct {
+	Region    string `toml:"region"`                 // AWS Region
+	AccessKey string `toml:"access_key"`             // Explicit AWS Access Key ID
+	SecretKey string `toml:"secret_key"`             // Explicit AWS Secret Access Key
+	RoleArn   string `toml:"role_arn"`               // Role ARN to assume
```

Comment:
These `toml` tags aren't actually supported because they are on an embedded, anonymous struct.
I've created a PR to add that support here - https://github.com/influxdata/toml/pull/6